When used as a submodule, PSA source files pick up the wrong config.h

### Description

When Mbed Crypto is used as a submodule in Mbed TLS, only the `config.h` of Mbed TLS should be used. But PSA source files use the ones in the submodule.

Steps to reproduce:
```
git clone https://github.com/ARMmbed/mbedtls
cd mbedtls
git submodule update
rm crypto/include/mbedtls/config.h
make
```

Expected behavior: since Mbed Crypto is used as a submodule in Mbed TLS, the `config.h` file under `crypto` should not be used.

Actual behavior:
```
…
  CC    psa_crypto.c
In file included from ../include/psa/crypto.h:25:0,
                 from psa_crypto.c:31:
../include/psa/crypto_platform.h:41:31: fatal error: ../mbedtls/config.h: No such file or directory
compilation terminated.
Makefile:146: recipe for target 'psa_crypto.o' failed
make[2]: *** [psa_crypto.o] Error 1
Makefile:152: recipe for target 'libmbedcrypto.a' failed
make[1]: *** [libmbedcrypto.a] Error 2
Makefile:19: recipe for target 'lib' failed
make: *** [lib] Error 2
```

Validation: to make sure that the wrong `config.h` is not used, we should do a build where its contents are replaced by something that will cause a build error, e.g. `#barf`.

### Issue request type

    [ ] Question
    [ ] Enhancement
    [x] Bug

--- a/include/psa/crypto_platform.h
+++ b/include/psa/crypto_platform.h
@@ -38,7 +38,7 @@
 /* Include the Mbed TLS configuration file, the way Mbed TLS does it
  * in each of its header files. */
 #if !defined(MBEDTLS_CONFIG_FILE)
-#include "../mbedtls/config.h"
+#include "mbedtls/config.h"
 #else
 #include MBEDTLS_CONFIG_FILE
 #endif

--- a/include/psa/crypto_sizes.h
+++ b/include/psa/crypto_sizes.h
@@ -45,7 +45,7 @@
 /* Include the Mbed TLS configuration file, the way Mbed TLS does it
  * in each of its header files. */
 #if !defined(MBEDTLS_CONFIG_FILE)
-#include "../mbedtls/config.h"
+#include "mbedtls/config.h"
 #else
 #include MBEDTLS_CONFIG_FILE
 #endif

--- a/include/psa/crypto_struct.h
+++ b/include/psa/crypto_struct.h
@@ -38,7 +38,7 @@
 /* Include the Mbed TLS configuration file, the way Mbed TLS does it
  * in each of its header files. */
 #if !defined(MBEDTLS_CONFIG_FILE)
-#include "../mbedtls/config.h"
+#include "mbedtls/config.h"
 #else
 #include MBEDTLS_CONFIG_FILE
 #endif

